src/main/java/org/dependencytrack/event/kafka/processor/MirrorVulnerabilityProcessor.java

```diff
@@ -105,25 +125,58 @@ public void process(final Record<String, Bom> record) {
         }
     }
 
-    public VulnerableSoftware mapAffectedPackageToVulnerableSoftware(final QueryManager qm, final Bom affectedPackage, String range, String bomRef) {
-        AtomicReference<String> purlStr = new AtomicReference<>();
-        AtomicReference<String> cpeStr = new AtomicReference<>();
-        if (!affectedPackage.getComponentsList().isEmpty()) {
-            affectedPackage.getComponentsList().stream().forEach(component -> {
-                if (component.getBomRef().equals(bomRef)) {
-                    purlStr.set(component.getPurl());
-                    cpeStr.set(component.getCpe());
+    public List<VulnerableSoftware> mapAffectedVersionToVulnerableSoftware(QueryManager qm, String version, String purlStr, String cpeStr) {
+        var vsList = new ArrayList<VulnerableSoftware>();
+        if (version != null) {
+            version = version.split("/")[1];
```

Comment:
Shouldn't `version` be just a single version, not a `vers` range? [`affects.versions`](https://cyclonedx.org/docs/1.4/json/#vulnerabilities_items_affects_items_versions) is an array, so

```json
[
  {
    "version": "vers:generic/0.10.0|0.10.1|0.11.0"
  }
]
```

should be

```json
[
  {
    "version": "0.10.0"
  },
  {
    "version": "0.10.1"
  },
  {
    "version": "0.11.0"
  }
]
```

or, alternatively

```json
[
  {
    "range": "vers:generic/0.10.0|0.10.1|0.11.0"
  }
]
```

I think this is more related the mirror-service, but the code here should not expect a range in the `version` field.